File: .sling/libs/web5/eqela/jk-net/0.38.0/jk.textprotocol.js
let JkTextprotocolTextProtocolConnection = function() {
	JkServerNetworkConnection.call(this);
	this.encoding = null;
};

JkTextprotocolTextProtocolConnection.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkServerNetworkConnection.prototype);
JkTextprotocolTextProtocolConnection.prototype.constructor = JkTextprotocolTextProtocolConnection;
JkTextprotocolTextProtocolConnection.prototype._t = {
	"JkServerNetworkConnection" : true,
	"JkTextprotocolTextProtocolConnection" : true
};
JkTextprotocolTextProtocolConnection.prototype._tobj = JkTextprotocolTextProtocolConnection;

JkTextprotocolTextProtocolConnection.NEW = function() {
	var v = new JkTextprotocolTextProtocolConnection;
	return v.CTOR_JkTextprotocolTextProtocolConnection();
};

JkTextprotocolTextProtocolConnection.prototype.CTOR_JkTextprotocolTextProtocolConnection = function() {
	this.encoding = null;
	if(JkServerNetworkConnection.prototype.CTOR_JkServerNetworkConnection.call(this) == null) {
		return null;
	}
	return this;
};

JkTextprotocolTextProtocolConnection.prototype.onDataReceived = function(data, size) {
	if(size < 1) {
		return;
	}
	var sb = JkLangBuffer.getSubBuffer(data, 0, size, false);
	if(!(sb != null)) {
		return;
	}
	var str = null;
	if(this.encoding == null) {
		str = JkLangString.forUTF8Buffer(sb);
	}
	else {
		str = JkLangString.forBuffer(data, this.encoding);
	}
	if(str == null) {
		return;
	}
	this.onTextReceived(str);
};

JkTextprotocolTextProtocolConnection.prototype.sendText = function(text) {
	if(text == null) {
		return;
	}
	this.sendData((JkLangString.toUTF8Buffer(text)), (-1));
};

JkTextprotocolTextProtocolConnection.prototype.onTextReceived = function(data) {};

JkTextprotocolTextProtocolConnection.prototype.getEncoding = function() {
	return this.encoding;
};

JkTextprotocolTextProtocolConnection.prototype.setEncoding = function(v) {
	this.encoding = v;
	return this;
};

JkTextprotocolTextProtocolConnection.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkTextprotocolTextProtocolConnection"] === true;
};

let JkTextprotocolTextLineProtocolConnection = function() {
	JkTextprotocolTextProtocolConnection.call(this);
	this.useCRLF = false;
};

JkTextprotocolTextLineProtocolConnection.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkTextprotocolTextProtocolConnection.prototype);
JkTextprotocolTextLineProtocolConnection.prototype.constructor = JkTextprotocolTextLineProtocolConnection;
JkTextprotocolTextLineProtocolConnection.prototype._t = {
	"JkTextprotocolTextProtocolConnection" : true,
	"JkServerNetworkConnection" : true,
	"JkTextprotocolTextLineProtocolConnection" : true
};
JkTextprotocolTextLineProtocolConnection.prototype._tobj = JkTextprotocolTextLineProtocolConnection;

JkTextprotocolTextLineProtocolConnection.NEW = function() {
	var v = new JkTextprotocolTextLineProtocolConnection;
	return v.CTOR_JkTextprotocolTextLineProtocolConnection();
};

JkTextprotocolTextLineProtocolConnection.prototype.CTOR_JkTextprotocolTextLineProtocolConnection = function() {
	this.useCRLF = false;
	if(JkTextprotocolTextProtocolConnection.prototype.CTOR_JkTextprotocolTextProtocolConnection.call(this) == null) {
		return null;
	}
	return this;
};

JkTextprotocolTextLineProtocolConnection.prototype.sendLine = function(text) {
	if(this.useCRLF) {
		this.sendText((JkLangString.safeString(text)) + ("\r\n"));
	}
	else {
		this.sendText((JkLangString.safeString(text)) + ("\n"));
	}
};

JkTextprotocolTextLineProtocolConnection.prototype.onTextReceived = function(data) {
	if(data == null) {
		return;
	}
	var str = data;
	if(JkLangString.endsWith(str, "\r\n")) {
		str = JkLangString.getSubString(str, 0, (JkLangString.getLength(str) - 2));
	}
	else if(JkLangString.endsWith(str, "\n")) {
		str = JkLangString.getSubString(str, 0, (JkLangString.getLength(str) - 1));
	}
	var nn = JkLangString.getIndexOfCharacter(str, '\n'.charCodeAt(), 0);
	if(nn < 0) {
		this.onLineReceived(str);
		return;
	}
	var array = JkLangString.split(str, '\n'.charCodeAt(), 0);
	if(array != null) {
		var n = 0;
		var m = array.length;
		for(n = 0 ; n < m ; n++) {
			var line = array[n];
			if(line != null) {
				if(JkLangString.endsWith(line, "\r")) {
					line = JkLangString.getSubString(line, 0, (JkLangString.getLength(line) - 1));
				}
				this.onLineReceived(line);
			}
		}
	}
};

JkTextprotocolTextLineProtocolConnection.prototype.onLineReceived = function(data) {};

JkTextprotocolTextLineProtocolConnection.prototype.getUseCRLF = function() {
	return this.useCRLF;
};

JkTextprotocolTextLineProtocolConnection.prototype.setUseCRLF = function(v) {
	this.useCRLF = v;
	return this;
};

JkTextprotocolTextLineProtocolConnection.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkTextprotocolTextLineProtocolConnection"] === true;
};

File: .sling/libs/sushi/eqela/jk-net/0.27.0/jk.textprotocol.lua
jk = jk or {}

jk.textprotocol = jk.textprotocol or {}

jk.textprotocol.TextProtocolConnection = _g.jk.server.NetworkConnection._create()
jk.textprotocol.TextProtocolConnection.__index = jk.textprotocol.TextProtocolConnection
_vm:set_metatable(jk.textprotocol.TextProtocolConnection, {
	__index = _g.jk.server.NetworkConnection
})

function jk.textprotocol.TextProtocolConnection._create()
	local v = _vm:set_metatable({}, jk.textprotocol.TextProtocolConnection)
	return v
end

function jk.textprotocol.TextProtocolConnection:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.textprotocol.TextProtocolConnection'
	self['_isType.jk.textprotocol.TextProtocolConnection'] = true
	self.encoding = nil
end

function jk.textprotocol.TextProtocolConnection:_construct0()
	jk.textprotocol.TextProtocolConnection._init(self)
	do _g.jk.server.NetworkConnection._construct0(self) end
	return self
end

function jk.textprotocol.TextProtocolConnection:onDataReceived(data, size)
	if size < 1 then
		do return end
	end
	do
		local sb = _g.jk.lang.Buffer:getSubBuffer(data, 0, size, false)
		if not (sb ~= nil) then
			do return end
		end
		do
			local str = nil
			if self.encoding == nil then
				str = _g.jk.lang.String:forUTF8Buffer(sb)
			else
				str = _g.jk.lang.String:forBuffer(data, self.encoding)
			end
			if str == nil then
				do return end
			end
			do self:onTextReceived(str) end
		end
	end
end

function jk.textprotocol.TextProtocolConnection:sendText(text)
	if text == nil then
		do return end
	end
	do self:sendData(_g.jk.lang.String:toUTF8Buffer(text), -1) end
end

function jk.textprotocol.TextProtocolConnection:onTextReceived(data)
end

function jk.textprotocol.TextProtocolConnection:getEncoding()
	do return self.encoding end
end

function jk.textprotocol.TextProtocolConnection:setEncoding(v)
	self.encoding = v
	do return self end
end

jk.textprotocol.TextLineProtocolConnection = _g.jk.textprotocol.TextProtocolConnection._create()
jk.textprotocol.TextLineProtocolConnection.__index = jk.textprotocol.TextLineProtocolConnection
_vm:set_metatable(jk.textprotocol.TextLineProtocolConnection, {
	__index = _g.jk.textprotocol.TextProtocolConnection
})

function jk.textprotocol.TextLineProtocolConnection._create()
	local v = _vm:set_metatable({}, jk.textprotocol.TextLineProtocolConnection)
	return v
end

function jk.textprotocol.TextLineProtocolConnection:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.textprotocol.TextLineProtocolConnection'
	self['_isType.jk.textprotocol.TextLineProtocolConnection'] = true
	self.useCRLF = false
end

function jk.textprotocol.TextLineProtocolConnection:_construct0()
	jk.textprotocol.TextLineProtocolConnection._init(self)
	do _g.jk.textprotocol.TextProtocolConnection._construct0(self) end
	return self
end

function jk.textprotocol.TextLineProtocolConnection:sendLine(text)
	if self.useCRLF then
		do self:sendText(_g.jk.lang.String:safeString(text) .. "\r\n") end
	else
		do self:sendText(_g.jk.lang.String:safeString(text) .. "\n") end
	end
end

function jk.textprotocol.TextLineProtocolConnection:onTextReceived(data)
	if data == nil then
		do return end
	end
	do
		local str = data
		if _g.jk.lang.String:endsWith(str, "\r\n") then
			str = _g.jk.lang.String:getSubString(str, 0, _g.jk.lang.String:getLength(str) - 2)
		elseif _g.jk.lang.String:endsWith(str, "\n") then
			str = _g.jk.lang.String:getSubString(str, 0, _g.jk.lang.String:getLength(str) - 1)
		end
		do
			local nn = _g.jk.lang.String:getIndexOfCharacter(str, 10, 0)
			if nn < 0 then
				do self:onLineReceived(str) end
				do return end
			end
			do
				local array = _g.jk.lang.String:split(str, 10, 0)
				if array ~= nil then
					local n = 0
					local m = _g.jk.lang.Vector:getSize(array)
					do
						n = 0
						while n < m do
							local line = array[n + 1]
							if line ~= nil then
								if _g.jk.lang.String:endsWith(line, "\r") then
									line = _g.jk.lang.String:getSubString(line, 0, _g.jk.lang.String:getLength(line) - 1)
								end
								do self:onLineReceived(line) end
							end
							do n = n + 1 end
						end
					end
				end
			end
		end
	end
end

function jk.textprotocol.TextLineProtocolConnection:onLineReceived(data)
end

function jk.textprotocol.TextLineProtocolConnection:getUseCRLF()
	do return self.useCRLF end
end

function jk.textprotocol.TextLineProtocolConnection:setUseCRLF(v)
	self.useCRLF = v
	do return self end
end
